components/trace_metric/src/metric.rs

```diff
@@ -61,12 +62,22 @@ impl Metric {
         })
     }
 
+    #[inline]
+    pub fn string(name: String, val: String, aggregator: Option<MetricAggregator>) -> Self {
+        Metric::String(MetricValue {
+            name,
+            val,
+            aggregator,
+        })
+    }
```

Comment:
String is not metric anymore... Just don't impl thing in a hack way without caring about suitable or not...